libs/asset-editor/src/lib/components/asset-editor-page/asset-editor-page.component.ts

```diff
@@ -1,50 +1,143 @@
+    filter((event) => event instanceof NavigationEnd),
+    startWith(() => undefined),
+    tap(() => {
+      const segments = (this.router.getCurrentNavigation() ?? this.router.lastSuccessfulNavigation)?.finalUrl?.root
+        .children?.['primary']?.segments;
+      if (segments !== undefined && segments.length > 0) {
+        this.activeTab = segments.pop()?.path as Tab;
+      }
+    })
+  );
+
+  public ngOnInit() {
+    this.subscriptions.add(this.activeItem$.subscribe());
+    this.form = buildForm();
+    this.loadAssetFromRouteParams();
+    this.subscriptions.add(
+      this.assetEditDetail$.subscribe((assetDetail) => {
+        if (this.assetId !== 'new') {
+          this.asset = assetDetail;
+        }
+        this.initializeForm();
+      })
+    );
+  }
+
+  public ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  public navigateToStart() {
+    this.routingService.navigateToRoot().then();
+  }
+
+  public initializeForm() {
+    this.form.reset();
+    this.form.controls.general.controls.titlePublic.setValue(this.asset?.titlePublic ?? null);
+  }
+
+  public openConfirmDialogForAssetDeletion(assetId: number) {
+    const dialogRef = this.dialogService.open<ConfirmDialogComponent>(ConfirmDialogComponent, {
+      data: {
+        text: 'confirmDelete',
+      },
+    });
+    dialogRef
+      .afterClosed()
+      .pipe(untilDestroyed(this))
+      .subscribe((hasConfirmed) => {
+        if (hasConfirmed) {
+          this.store.dispatch(actions.deleteAsset({ assetId }));
+        }
+      });
+  }
+
+  public canDeactivate(targetRoute: RouterStateSnapshot): boolean | Observable<boolean> {
+    if (!this.form.dirty || targetRoute.url.includes(this.assetId.toString())) {
+      return true;
+    }
+    const dialogRef = this.dialogService.open<ConfirmDialogComponent>(ConfirmDialogComponent, {
+      data: {
+        text: 'confirmDiscardChanges',
+      },
     });
+    return dialogRef.afterClosed();
   }
 
-  public canLeave(): Observable<boolean> {
-    if (!this.tabPageBridgeService.tabPage) return of(true);
-    return this.tabPageBridgeService.tabPage.canLeave();
+  private loadAssetFromRouteParams() {
+    this.subscriptions.add(
+      this.route.paramMap
+        .pipe(withLatestFrom(this.store.select(fromAppShared.selectCurrentAsset)), take(1))
+        .subscribe(([params, currentAsset]) => {
+          const assetIdFromParam = params.get('assetId');
+          if (!assetIdFromParam || !parseInt(assetIdFromParam)) {
+            return;
+          }
+          const assetId = parseInt(assetIdFromParam);
+
+          if (currentAsset?.assetId === assetId) {
+            this.assetId = currentAsset.assetId;
+            this.asset = currentAsset;
+          } else {
+            this.assetId = assetId;
+            this.store.dispatch(actions.loadAsset({ assetId }));
+          }
+        })
+    );
   }
+
+  protected readonly Tab = Tab;
 }
+
+function buildForm() {
```

Comment:
Arrow functions > regular functions 😊 
```suggestion
const buildForm = () => {
```

Reply:
why? is this only for legibiility in this case? or are there other reasons?

Reply:
My main two reasons are for standardization and because I find them easier to reason about (among other things, regular functions are subject to [hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting), which is also the reason you should use `let` over `var`).

Another thing is that `function`s have their own [`this`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) binding, while arrow functions don't. This, to me, signals that whenever I see a `function`, that its `this` has a special meaning.